mathics/builtin/randomnumbers.py

```diff
@@ -33,17 +65,60 @@ def get_random_state():
 
 def set_random_state(state):
     if state is None:
-        random.seed()
+        random_seed()
     else:
         state = hex(state)[2:]  # drop leading "0x"
         state = state.rstrip('L')
         state = state.encode('ascii')
         state = binascii.a2b_hex(state)
         state = pickle.loads(state)
-        random.setstate(state)
+        random_set_state(state)
+
+
+if _numpy:
+    def _from_array(a, new_element, d=1):
+        if len(a.shape) == d:
+            leaves = [new_element(x) for x in a]
+        else:
+            leaves = [_from_array(e, new_element, d) for e in a]
+        return Expression('List', *leaves)
+
+    def _stack_array(a):
+        # numpy.stack with axis=-1 stacks arrays along the most inner axis:
+        # e.g. numpy.stack([ [1, 2], [3, 4] ], axis=-1)
+        # gives: array([ [1, 3], [2, 4] ])
+        # e.g. numpy.stack([ [[1, 2], [3, 4]], [[4, 5], [6, 7]] ], axis=-1)
+        # gives: array([[[1, 4], [2, 5]], [[3, 6], [4, 7]]])
+        return numpy.stack(a, axis=-1)
+else:
+    from itertools import chain
+
+    def _create_array(size, f):
+        if size is None or len(size) == 0:
+            return f()
+        else:
+            return [_create_array(size[1:], f) for _ in range(size[0])]
+
+    def _from_array(a, new_element, d=1):
+        e = a[0]
+        depth = 1
+        while depth <= d and isinstance(e, list):
+            e = e[0]
+            depth += 1
+        if d == depth:
+            leaves = [new_element(x) for x in a]
+        else:
+            leaves = [_from_array(e, new_element, d) for e in a]
+        return Expression('List', *leaves)
+
+    def _stack_array(a):
+        if not isinstance(a[0], list):
+            return list(chain(a))
+        else:
+            return [_stack_array([x[i] for x in a]) for i in range(len(a[0]))]
```

Comment:
I wonder if these functions might be useful for other things that could use numpy, e.g. linear algebra routines. There's a short discussion on the mailing list see https://groups.google.com/forum/#!topic/mathics-devel/6btOa-5q87E but I think the best way to proceed would be a separate module e.g. `mathics/builtin/numpy_utils.py`. What do you think?